internal/nexus/nexus-abstract.go

```diff
@@ -65,6 +67,14 @@ func (w *NexusProjectActiveWatcher) GetSpec() *projectActiveWatcherv1.ProjectAct
 	return &w.Spec
 }
 
+func (w *NexusProjectActiveWatcher) GetAnnotations() map[string]string {
```

Comment:
@badhrinathpa possible issue here. Try this:

func (w *NexusProjectActiveWatcher) GetAnnotations() map[string]string {
	return (*nexus.ProjectactivewatcherProjectActiveWatcher)(w).GetAnnotations()
}

func (w *NexusProjectActiveWatcher) SetAnnotations(annotations map[string]string) {
	(*nexus.ProjectactivewatcherProjectActiveWatcher)(w).SetAnnotations(annotations)
}

Reply:
The issue was because of an extra line added during merge
`if watcherObj.GetSpec().StatusIndicator == projectActiveWatcherv1.StatusIndicationIdle && watcherObj.GetSpec().Message == "Created" {
-               // This is a rerun of an event we already processed - no more processing required
-               log.Infof("Watch %s for project %s already provisioned", watcherObj.DisplayName(), project.DisplayName())
-               return nil
-       }`

Reply:
If I'm looking at the correct Nexus source code, there is some guarding in the SetAnnotations and GetAnnotations to cover the case where the annotations are currently empty, and create them. Maybe our code always sees the objects with the annotations already set, so it works. However, I still wonder if calling the original funcs would be safer.

Reply:
Done.